gammapy/visualization/datasets.py

```diff
@@ -0,0 +1,175 @@
+    ``legend_kwargs`` only applies if it's given in the last function call with ``legend=True``::
+
+        plt.rc('legend', columnspacing=1, fontsize=9)
+        plot_spectrum_datasets_off_regions(datasets1, ax, legend=True, edgecolor='cyan')
+        plot_spectrum_datasets_off_regions(datasets2, ax, legend=True, legend_kwargs=dict(ncol=2))
+        plot_spectrum_datasets_off_regions(datasets3, ax, legend=False, edgecolor='magenta')
+    """
+    from matplotlib.legend_handler import HandlerPatch, HandlerTuple
+    from matplotlib.patches import CirclePolygon, Patch
+
+    if ax is None:
+        ax = plt.subplot(projection=datasets[0].counts_off.geom.wcs)
+
+    legend = legend or legend is None and len(datasets) <= 10
+    legend_kwargs = legend_kwargs or {}
+    handles, labels = [], []
+
+    prop_cycle = kwargs.pop("prop_cycle", plt.rcParams["axes.prop_cycle"])
+
+    for props, dataset in zip(prop_cycle(), datasets):
+        plot_kwargs = kwargs.copy()
+        plot_kwargs["facecolor"] = "None"
+        plot_kwargs.setdefault("edgecolor")
+        plot_kwargs.update(props)
+
+        dataset.counts_off.plot_region(ax, **plot_kwargs)
+
+        # create proxy artist for the custom legend
+        if legend:
+            handle = Patch(**plot_kwargs)
+            handles.append(handle)
+            labels.append(dataset.name)
+
+    if legend:
+        legend = ax.get_legend()
+        if legend:
+            handles = legend.legendHandles + handles
+            labels = [text.get_text() for text in legend.texts] + labels
+
+        handles = [(handle, handle) for handle in handles]
+        tuple_handler = HandlerTuple(ndivide=None, pad=0)
+
+        def patch_func(
+            legend, orig_handle, xdescent, ydescent, width, height, fontsize
+        ):
+            radius = width / 2
+            return CirclePolygon((radius - xdescent, height / 2 - ydescent), radius)
+
+        patch_handler = HandlerPatch(patch_func)
+
+        legend_kwargs.setdefault("handletextpad", 0.5)
+        legend_kwargs["handler_map"] = {Patch: patch_handler, tuple: tuple_handler}
+        ax.legend(handles, labels, **legend_kwargs)
+
+    return ax
+
+
+def plot_npred_signal(
+    dataset,
+    ax=None,
+    model_names=None,
+    stack=False,
+    region=None,
+    plot_background=True,
+    **kwargs,
+):
+    """
+    Plot the npred_signal of the models of a dataset.
+
+    Parameters
+    ----------
+    dataset : an instance of `~gammapy.datasets.dataset`
+        The dataset from which to plot the npred_signal
+    ax : `~matplotlib.axes.Axes`
+        Axis object to plot on.
+    model_names : list of str
+        The list of model for which the npres_signal is plotted. If set to None, it will take all the models
+        of the `dataset
+    stack : bool
+        Whether to sum the npred_signal of all the model.
+    region: `~regions.Region` or `~astropy.coordinates.SkyCoord`
+        Region.
```

Comment:
Region used to reproject predicted counts. If None, use full dataset geometry.
Default is None.